runtime/lua/vim/filetype/detect.lua

```diff
@@ -0,0 +1,362 @@
+local M = {}
+
+---@private
+local function count_lines(bufnr)
+  return vim.api.nvim_buf_line_count(bufnr)
+end
+
+---@private
+local function get_lines(bufnr, start_lnum, end_lnum)
+  end_lnum = end_lnum or start_lnum
+  local lines = vim.api.nvim_buf_get_lines(bufnr, start_lnum - 1, end_lnum, false)
+  return table.concat(lines) or ""
+end
+
+---@private
+local function iter_lines(bufnr, start_lnum, end_lnum)
+  end_lnum = end_lnum or start_lnum
+  return ipairs(vim.api.nvim_buf_get_lines(bufnr, start_lnum - 1, end_lnum, false))
+end
+
+function M.asm()
+
+end
+
+function M.asm_syntax()
+
+end
+
+function M.bas()
+
+end
+
+function M.bindzone()
+
+end
+
+function M.btm(_, bufnr)
+  if vim.g.dosbatch_syntax_for_btm ~= nil and vim.g.dosbatch_syntax_for_btm ~= 0 then
+    vim.bo[bufnr].filetype = "dosbatch"
+  else
+    vim.bo[bufnr].filetype = "btm"
+  end
+end
+
+function M.cfg()
+
+end
+
+function M.change()
+
+end
+
+function M.csh()
+
+end
+
+function M.dat()
+
+end
+
+function M.dep3patch()
+
+end
+
+function M.dtrace()
+
+end
+
+function M.e()
+
+end
+
+-- This function checks for valid cl syntax in the first five lines.
+-- Look for either an opening comment, '#', or a block start, '{'.
+-- If not found, assume SGML.
+function M.ent(_, bufnr)
+  for _, line in iter_lines(bufnr, 1, 5) do
+    if line:find("^%s*[#{]") then
+      vim.bo[bufnr].filetype = "cl"
+      return
+    elseif not line:find("^%s*$") then
+      -- Not a blank line, not a comment, and not a block start,
+      -- so doesn't look like valid cl code.
+      break
+    end
+  end
+  vim.bo[bufnr].filetype = "dtd"
+end
+
+function M.euphoria_check(_, bufnr)
+  if vim.g.filetype_euphoria ~= nil then
```

Comment:
```suggestion
  if vim.g.filetype_euphoria then
```
(and similarly everywhere)